frontend/components/details/ModuleHTML.tsx

```diff
@@ -0,0 +1,60 @@
+import { HTMLReactParserOptions } from "html-react-parser";
+import { Element } from "html-react-parser/node_modules/domhandler/lib/node";
+import { FC, ReactElement, useEffect, useState } from "react";
+
+interface ModuleHTMLProps {
+  html: string;
+}
+
+const ModuleHTML: FC<ModuleHTMLProps> = ({ html }: ModuleHTMLProps) => {
+  const [renderedHTML, setRenderedHTML] = useState(() => (
+    <div dangerouslySetInnerHTML={{ __html: html }} />
+  ));
+
+  useEffect(() => {
+    Promise.all([import("html-react-parser"), import("@/components/details/ModuleLink")]).then(
+      ([{ default: parse, domToReact, attributesToProps }, { default: ModuleLink }]) => {
```

Comment:
https://nextjs.org/docs/advanced-features/dynamic-import
Dynamic imports mean smaller bundle for initial page loads.